optimade/filtertransformers/mongo.py

```diff
@@ -208,30 +219,82 @@ def property_zip_addon(self, arg):
 
     def _recursive_expression_phrase(self, arg):
         """Helper function for parsing `expression_phrase`. Recursively sorts out
-        the correct precedence for `$and`, `$or` and `$nor`.
+        the correct precedence for `$not', '$and` and `$or`.
 
         """
-        if len(arg) == 1:
-            # without NOT
-            return arg[0]
 
-        # handle the case of {"$not": {"$or": [expr1, expr2]}} using {"$nor": [expr1, expr2]}.
-        if "$or" in arg[1]:
-            return {"$nor": self._recursive_expression_phrase([arg[1]["$or"]])}
+        def handle_not_and(arg):
+            """Handle the case of `{"$not": {"$and": [expr1, expr2]}}`
+            which is equal to `{"or": [{"$not": expr1}, {"$not": expr2}]}`}
+            We have to check for the special case in which the "and" was created by a previous NOT
+            e.g `NOT (NOT ({"a": {"$eq" : 6}})) -> NOT({$and:[{"a": {"$ne" : 6}},{"a": {"$ne": None}}]})`
+
+            """
+
+            expr1 = arg["$and"][0]
+            expr2 = arg["$and"][1]
+            key = list(expr1.keys())[0]
+            if expr1.keys() == expr2.keys():
+                if expr1.get(key) == {"$ne": None}:
+                    return self._recursive_expression_phrase(["NOT", expr2])
+                elif expr2.get(key) == {"$ne": None}:
+                    return self._recursive_expression_phrase(["NOT", expr1])
```

Comment:
Might be overkill here, but you could cut a line by using `itertools.permutations` and something like:

```suggestion
            if expr1.keys() == expr2.keys():
                for e, f in itertools.permutations((expr1, expr2)):
                    if e.get(key) == {"$ne": None}:
                        return self._recursive_expression_phrase(["NOT", f])
```

might be marginally clearer.